[nativert] Move ParallelGraphExecutor to PyTorch core

henryoier opened the pull request:
Summary: `ParallelGraphExecutor` inherits from `GraphExecutorBase` and executes all nodes in the graph in a parallel manner

Test Plan:
CI

Rollback Plan:

Differential Revision: D77088996

henryoier commented:
@pytorchbot label "topic: not user facing"

dolpm commented on torch/nativert/executor/ParallelGraphExecutor.h:
```diff
@@ -0,0 +1,94 @@
+#pragma once
+
+#include <c10/util/Semaphore.h>
+#include <torch/nativert/executor/GraphExecutorBase.h> // @manual
```
can get rid of ```// @manual``` now

henryoier committed:
[nativert] Move ParallelGraphExecutor to PyTorch core (#156751)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/156751

`ParallelGraphExecutor` inherits from `GraphExecutorBase` and executes all nodes in the graph in a parallel manner

Test Plan:
CI

Rollback Plan:

Reviewed By: zhxchen17, dolpm

Differential Revision: D77088996
